subdomains/blog/_posts/2020-11-25-command-spotlight-volta-run.md

```diff
@@ -0,0 +1,41 @@
+---
+layout: post
+title:  "Command Spotlight: volta run"
+date:   2020-11-23 08:00:00 -0700
+categories: [spotlight, commands]
+author: Chuck Pierce
+excerpt_separator: <!--more-->
+---
+
+Volta is designed to be invisible: You type a command like `npm start`, and Volta automatically detects and runs the right version of npm. While that works great most of the time, occasionally you need direct control over the tool versions that are used. Maybe you are trying to reproduce reported bug or testing how your project works with a different npm version. Whatever the reason, **`volta run` lets you take control of your tool selection right from the command line**.
+<!--more-->
+
+Let's check out some examples of how you can use `volta run`:
+
+```
+volta run --node lts --npm 7 npm start
+```
+
+This will run the command `npm start` using the most recent LTS Node version and major version 7 of npm.
+
+```
+volta run --yarn 1.22.1 yarn add typescript
+```
+
+This will run `yarn add typescript` using Yarn 1.22.1 and whichever Node version Volta would normally find (probably from your `package.json` file).
+
+```
+volta run --node latest --bundled-npm --env NPM_CONFIG_LOGLEVEL=silly npm install
+```
+
+This will run `npm install` using the latest Node, the npm version bundled with that Node, and with the environment variable `NPM_CONFIG_LOGLEVEL` set to `silly`.
+
+```
```

Comment:
I wonder if instead of this last one, which is pretty esoteric, it might be more interesting to talk about how it can be pretty convenient to run tests over a matrix of Node or npm versions in CI systems like GitHub Actions by passing a CI environment variable as a parameter to `volta run`:

```yaml
# or whatever... I forget the Actions syntax exactly
script:
    # or whatever... I forget Yaml syntax exactly 😅
    - volta run --node $NODE_VERSION --npm $NPM_VERSION npm start
```

Reply:
Ohh, I like that! It's an example with an explicit use-case. Let me dig up the GitHub actions syntax and put something together.

Reply:
Updated to use CI as an example.